helm-charts/splunk-otel-collector/templates/daemonset.yaml

```
@@ -404,11 +404,16 @@ spec:
           readOnly: true
         {{- end }}
         {{- end }}
+        - mountPath: /usr/lib/splunk-otel-collector/agent-bundle/run/collectd
+          name: run-collectd
+          readOnly: false
         {{- if $agent.extraVolumeMounts }}
         {{- toYaml $agent.extraVolumeMounts | nindent 8 }}
         {{- end }}
       terminationGracePeriodSeconds: {{ .Values.terminationGracePeriodSeconds }}
       volumes:
+      - name: run-collectd
+        emptyDir: {}
```

How about adding a limit? I assume we are only writing collectd config files in this directory.
```suggestion
        emptyDir:
          sizeLimit: 500Mi
```

---

good call, though that would be a lot of config files and they should be cleanly removed on receiver shutdown.